plugins/coverimage.koplugin/main.lua

```diff
@@ -24,6 +25,17 @@ function CoverImage:_restore()
     return self.enabled and self.restore
 end
 
+-- check if settings of lastfile prohibits creation of cover image
+function CoverImage:excluded()
+    local lastfile = G_reader_settings:readSetting("lastfile")
+    local exclude_ss = false -- consider it not excluded if there's no docsetting
+    if DocSettings:hasSidecarFile(lastfile) then
+        local doc_settings = DocSettings:open(lastfile)
+        exclude_ss = doc_settings:readSetting("exclude_cover_image")
```

Comment:
This is re-opening/parsing the metatada.zzz.lua, and might be costly. In most (or all, not sure about onCloseDocument) your contexts, you already have self.ui.odc_settings or doc_settings provided, so you can just `readSetting("exclude_cover_image")` on it - without having to re-open/parse it.

Reply:
In line 34:
` exclude_ss = self.ui.doc_settings:readSetting("exclude_cover_image") ` brings `./luajit: plugins/coverimage.koplugin/main.lua:35: attempt to index field 'ui' (a nil value)`

and 
`        exclude_ss = DocSettings:readSetting("exclude_cover_image")`
 brings up `./luajit: frontend/docsettings.lua:170: attempt to index field 'data' (a nil value)`

So i have combined the old method with self.ui.doc_settings

Reply:
For the 2nd one:
`exclude_ss = DocSettings:readSetting("exclude_cover_image")`
`DocSettings` is a module/class, not an instance, so it has no data, nothing to readerSetting from :)